packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -91,13 +121,57 @@ export class CodeBuildStartBuild extends sfn.TaskStateBase {
+        BuildspecOverride: this.props.overrides?.buildSpec?.toBuildSpec(),
+        BuildStatusConfigOverride: sourceConfig?.sourceProperty.buildStatusConfig,
+        ComputeTypeOverride: this.props.overrides?.environment?.computeType,
+        DebugSessionEnabled: this.props.overrides?.debugSessionEnabled,
+        EncryptionKeyOverride: this.props.overrides?.encryptionKey?.keyArn,
+        EnvironmentTypeOverride: this.props.overrides?.environment?.buildImage?.type,
+        EnvironmentVariablesOverride: this.props.overrides?.environmentVariables ?
+          this.serializeEnvVariables(this.props.overrides?.environmentVariables!) :
+          (this.props.environmentVariablesOverride
+            ? this.serializeEnvVariables(this.props.environmentVariablesOverride)
+            : undefined),
+        GitCloneDepthOverride: sourceConfig?.sourceProperty.gitCloneDepth,
+        GitSubmodulesConfigOverride: sourceConfig?.sourceProperty.gitSubmodulesConfig,
+        IdempotencyToken: this.props.overrides?.idempotencyToken,
+        ImageOverride: this.props.overrides?.environment?.buildImage?.imageId,
+        ImagePullCredentialsTypeOverride: this.props.overrides?.environment?.buildImage?.imagePullPrincipalType,
+        InsecureSslOverride: sourceConfig?.sourceProperty.insecureSsl,
+        PrivilegedModeOverride: this.props.overrides?.environment?.privileged,
+        RegistryCredentialOverride: this.props.overrides?.environment?.buildImage?.secretsManagerCredentials ? {
+          credentialProvider: 'SECRETS_MANAGER',
+          credential: this.props.overrides!.environment!.buildImage!.secretsManagerCredentials.secretArn,
+        } : undefined,
+        ReportBuildStatusOverride: sourceConfig?.sourceProperty.reportBuildStatus,
+        SecondaryArtifactsOverride: this.props.overrides?.secondaryArtifacts?.map(artifact =>
+          artifact.bind(this.props.project.stack, this.props.project).artifactsProperty,
+        ),
+        SecondarySourcesOverride: secondarySources?.map(source => source.sourceProperty),
+        SecondarySourcesVersionOverride: secondarySources?.map(source => {
+          return {
+            sourceIdentifier: source.sourceProperty.sourceIdentifier,
+            sourceVersion: source.sourceVersion,
+          };
+        }),
+        ServiceRoleOverride: this.props.overrides?.role?.roleArn,
+        SourceAuthOverride: sourceConfig?.sourceProperty.auth,
+        SourceLocationOverride: sourceConfig?.sourceProperty.location,
+        SourceTypeOverride: this.props.overrides?.source?.type,
+        SourceVersion: sourceConfig?.sourceVersion,
+        TimeoutInMinutesOverride: this.props.overrides?.timeout?.toMinutes(),
+        // below override properties not supported yet
+        // CacheOverride:
+        // CertificateOverride:
+        // LogsConfigOverride:
+        // QueuedTimeoutInMinutesOverride:
```

Comment:
rather than this can we create an issue and point to it instead? - would be a lot cleaner as well as capture the rationale for why we looked past them for now.